src/rust/engine/src/externs/mod.rs

```diff
@@ -66,7 +66,7 @@ pub struct PyFailure(pub Failure);
 impl PyFailure {
     fn get_error(&self, py: Python) -> PyErr {
         match &self.0 {
-            Failure::Throw { val, .. } => val.into_py(py),
+            Failure::Throw { val, .. } => PyErr::from_value(val.as_ref().as_ref(py)),
```

Comment:
I'm assuming this change goes along with the change in src/rust/engine/src/python.rs?
But I don't follow all the `.as_ref()` calls and what they do :p (not saying I have to.. guess my question is: is this clear to folks who speak Rust?)

Reply:
> I'm assuming this change goes along with the change in src/rust/engine/src/python.rs?

Yea, exactly.

> But I don't follow all the `.as_ref()` calls and what they do :p (not saying I have to.. guess my question is: is this clear to folks who speak Rust?)

Relatively, yes: there are just multiple layers of wrapping types around a Python object, some of which you need to be holding the Gil (the `py: Python` token represents the Gil being held) to be allowed to dereference.